### PR TITLE
Add support for preview tokens

### PR DESCRIPTION
**Pull request by dan-alcantara-neeva**
* Change the `AuthenticatedApolloWrapper` so that it also manages a `PreviewSessionToken`.  If the `LoginToken` is empty when an Apollo call is made, we try to ensure that a `PreviewSessionToken` is available by asking the WebLayer `Browser`'s `CookieManager` for it.  If it doesn't exist, we fire off a network request to create one and persist it.

* Generalize the call from the `IncognitoApolloWrapper` that checks for an empty token and requests one, putting it in the `SessionToken` class so that we can use it to get Preview session tokens.
  * The `LoginToken` overrides this to avoid trying to fetch a token because login tokens require the user to manually input their password and log in.
  * Update the `SessionTokenTest` so that it checks the new `getOrFetchCookie()` function that used to live in `IncognitoApolloWrapper`.

* Fix the nullability of the returned values in the `ApolloWrapper` classes.

* Add the `AuthenticatedApolloWrapperTest` to mimic the `IncognitoApolloWrapperTest`, which was updated to account for the fact that the `SessionToken` now fires its own network request if it doesn't have a valid token.

### Pointers to help with review
* [Android and Neeva Tokens doc](https://docs.google.com/document/d/1rRStjSXig6HfbaXPl4cXbJH_jGkRpTWdTIGsY7G9Gx0/edit#heading=h.sg0ob8ttl9ru)

### How was this tested?
- [ ] I tested this manually
- [X] I added automated tests
- [X] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes

### Manual test cases
If manually tested, describe the different scenarios you followed to ensure
that your changes work.  Make sure that you've tested as many pathways as
possible -- not just the [happy path](https://en.wikipedia.org/wiki/Happy_path).

### Screenshots and screen recordings
Images or videos showing the effect of your PR.  Examples include:
* For new Composables, add images of `@Preview`s you have added for both light
  and dark mode
* Before and after screenshots showing the effect of your styling changes
* Videos showing you performing the flow that you have changed

## Issues addressed
* Link(s) to Github issue(s) (if applicable)

### CODE REVIEW

File: app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt [hunk 1]
```diff
@@ -6,16 +6,37 @@ package com.neeva.app.apollo
...
-        return !userMustBeLoggedIn || !loginToken.isEmpty()
+        val previewCookieValue = if (loginToken.isEmpty()) {
+            previewSessionToken.getOrFetchCookie()
+        } else {
+            // The backend gives the login cookie precedence over the preview cookie, ignoring the
+            // preview cookie entirely if the login cookie is set.
+            null
+        }
+
+        return !userMustBeLoggedIn || !loginToken.isEmpty() || !previewCookieValue.isNullOrEmpty()
```
Comment (Kokobe) [inline app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt:34]: Just a suggestion for readability/understanding, maybe use this variable in the return:
```
val hasValidToken = !loginToken.isEmpty() || !previewCookieValue.isNullOrEmpty()
return !userMustBeLoggedIn || hasValidToken
```

Also realizing that "Token" and "CookieValue" have been used interchangeably everywhere

File: app/src/test/java/com/neeva/app/apollo/IncognitoApolloWrapperTest.kt [hunk 2]
```diff
@@ -49,36 +52,38 @@ class IncognitoApolloWrapperTest : BaseTest() {
...
+            runBlocking { apolloWrapper.prepareForOperation(userMustBeLoggedIn = false) }
+        ).isTrue()
+        coVerify(exactly = 1) { incognitoSessionToken.getOrFetchCookie() }
 
-        verify(exactly = 1) { incognitoSessionToken.requestNewCookie() }
-        coVerify(exactly = 1) { incognitoSessionToken.waitForRequest() }
+        expectThat(
+            runBlocking { apolloWrapper.prepareForOperation(userMustBeLoggedIn = true) }
+        ).isFalse()
+        coVerify(exactly = 2) { incognitoSessionToken.getOrFetchCookie() }
```
[the remark this answers is not in the document]
Comment (dan-alcantara-neeva) [reply app/src/test/java/com/neeva/app/apollo/IncognitoApolloWrapperTest.kt:81]: Happens inside of the `prepareForOperation`  call up top.  The counters don't reset unless you explicitly do it

**Review by Kokobe — APPROVED** [review]

File: app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt (hunk 1 above)
Comment (dan-alcantara-neeva) [reply app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt:34]: Yeah, seems to have been carried forward from the iOS side of things.  We can rename them all to cookies in a follow-up PR so it's obvious that there are no functional changes.

File: app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt [hunk 3]
```diff
@@ -6,16 +6,37 @@ package com.neeva.app.apollo
 
 import com.neeva.app.NeevaConstants
 import com.neeva.app.userdata.LoginToken
+import com.neeva.app.userdata.PreviewSessionToken
+import org.chromium.weblayer.Browser
 
 /** Authenticated [BaseApolloWrapper] that sends the user token if the user is signed in. */
 open class AuthenticatedApolloWrapper(
```
[the remark this answers is not in the document]
Comment (dan-alcantara-neeva) [reply app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt:13]: Will look into moving it directly into the SpaceStore as a separate PR.  It's currently injected everywhere, so it'd muddle the changes in this PR.

File: app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt (hunk 1 above)
Comment (dan-alcantara-neeva) [reply app/src/main/java/com/neeva/app/apollo/AuthenticatedApolloWrapper.kt:34]: Talked offline; will change this up in a follow-up PR.

**Pull request merged**